stuffer/s2n_stuffer.c

```diff
@@ -437,7 +442,13 @@ int s2n_stuffer_shift(struct s2n_stuffer *stuffer)
     POSIX_ENSURE_REF(stuffer);
     struct s2n_stuffer copy = *stuffer;
     POSIX_GUARD(s2n_stuffer_rewrite(&copy));
-    uint8_t *data = stuffer->blob.data + stuffer->read_cursor;
+    uint8_t *data = stuffer->blob.data;
+
+    /* Prevent null arithmetic if stuffer->read_cursor is zero */
```

Comment:
Let's document "why" as opposed to "what", and same for the other comments. So something like
> Adding 0 to a null value is undefined behavior. This if statement prevents undefined behavior in the case where `stuffer->blob.data` is `NULL` and `suffer->read_cursor` is `0`.

Reply:
Nit: line 445 is conceptually part of the same block as the if statement, so it should probably be below the comment and right above the if statement.

Reply:
I will update and relocate the comment in the next commit.